### PR TITLE
Add markup for Foreman Amazon EC2 compute resource

### PR DESCRIPTION
**Pull request by melcorr**
* Add ifevals for Satellite-specific
* Add ifevals for Foreman-specific

### CODE REVIEW

File: guides/doc-Provisioning_Guide/topics/Cloud-Amazon_EC2.adoc [hunk 1]
```diff
@@ -1,25 +1,38 @@
...
   * Synchronized content repositories for Red{nbsp}Hat Enterprise Linux 7. For more information, see link:/html/content_management_guide/importing_red_hat_content#Importing_Red_Hat_Content-Synchronizing_Red_Hat_Repositories[Synchronizing Red{nbsp}Hat Repositories] in the _Content Management Guide_.
+endif::[]
   * A {SmartProxyServer} managing a network in your EC2 environment. Use a Virtual Private Cloud (VPC) to ensure a secure network between the hosts and the {SmartProxyServer}.
   * An Amazon Machine Image (AMI) for image-based provisioning.
+ifeval::["{Build}" == "satellite"]
   * An activation key for host registration. For more information, see link:/html/content_management_guide/managing_activation_keys#Managing_Activation_Keys-Creating_an_Activation_Key[Creating An Activation Key] in the _Content Management_ guide.
+endif::[]
+ifeval::["{Build}" == "foreman"]
+  * If the Katello plugin is installed, an activation key for host registration. For more information, see link:/html/content_management_guide/managing_activation_keys#Managing_Activation_Keys-Creating_an_Activation_Key[Creating An Activation Key] in the _Content Management_ guide.
+endif::[]
```
Comment (lzap) [inline guides/doc-Provisioning_Guide/topics/Cloud-Amazon_EC2.adoc]: Looks probably better than my version (libvirt PR). Go ahead and compare if you like anything from my version but this is fine.

File: guides/doc-Provisioning_Guide/topics/Cloud-Amazon_EC2.adoc [hunk 2]
```diff
@@ -1,25 +1,38 @@
...
+Amazon Web Services uses time settings as part of the authentication process. Ensure that {ProjectServer}'s time is correctly synchronized. Ensure that an NTP service, such as `ntpd` or `chronyd`, is running properly on the {ProjectServer}. Failure to provide the correct time to Amazon Web Services can lead to authentication failures.
+ifeval::["{Build}" == "satellite"]
+For more information about synchronizing time in {Project}, see link:link:/html/installing_satellite_server_from_a_connected_network/installing_satellite_server#synchronizing_time[Synchronizing Time] in _Installing {ProjectServer} from a Connected Network_.
+endif::[]
```
Comment (lzap) [inline guides/doc-Provisioning_Guide/topics/Cloud-Amazon_EC2.adoc]: These links are invalid now, but will be valid in the future. I'd not hide them.

File: guides/doc-Provisioning_Guide/topics/Cloud-Amazon_EC2.adoc [hunk 3]
```diff
@@ -29,7 +42,10 @@ To add an Amazon EC2 connection, complete the following steps:
 . In the *Name* field, enter a name to identify the Amazon EC2 compute resource.
 . From the *Provider* list, select *EC2*.
 . In the *Description* field, enter information that helps distinguish the resource for future use.
-. From the *HTTP proxy* list, select a HTTP proxy to connect to external API services. You must add HTTP proxies to {Project} before you can select a proxy from this list. For more information, see link:link:/html/installing_satellite_server_from_a_connected_network/performing_additional_configuration_on_satellite_server#configuring_satellite_http_proxy[Configuring {ProjectServer} with HTTP Proxy] in _Installing {ProjectServer} from a Connected Network_.
+. From the *HTTP proxy* list, select a HTTP proxy to connect to external API services. You must add HTTP proxies to {Project} before you can select a proxy from this list.
+ifeval::["{Build}" == "satellite"]
+For more information, see link:link:/html/installing_satellite_server_from_a_connected_network/performing_additional_configuration_on_satellite_server#configuring_satellite_http_proxy[Configuring {ProjectServer} with HTTP Proxy] in _Installing {ProjectServer} from a Connected Network_.
+endif::[]
```
Comment (lzap) [inline guides/doc-Provisioning_Guide/topics/Cloud-Amazon_EC2.adoc]: Ditto.

File: guides/doc-Provisioning_Guide/topics/Cloud-Amazon_EC2.adoc [hunk 4]
```diff
@@ -247,15 +274,27 @@ EOS
...
 .. In the *Name* field, enter `remote_execution_ssh_keys`. In the corresponding *Value* field, enter the output of `cat /usr/share/foreman/.ssh/id_rsa_foreman_proxy.pub`.
 .. In the *Name* field, enter `remote_execution_ssh_user`. In the corresponding *Value* field, enter `ec2-user`.
+ifeval::["{Build}" == "satellite"]
 .. In the *Name* field, enter `activation_keys`. In the corresponding *Value* field, enter your activation key.
+endif::[]
+ifeval::["{Build}" == "foreman"]
+.. If you use the Katello plugin, in the *Name* field, enter `activation_keys`. In the corresponding *Value* field, enter your activation key.
+endif::[]
 . Click *Submit* to save the changes.
```
Comment (lzap) [inline guides/doc-Provisioning_Guide/topics/Cloud-Amazon_EC2.adoc]: It is actually more complicated - remote_execution stuff only appears when Remote Execution plugin is installed. Probably just keep it as-is and add a NOTE that will explain some might be missing when ReX or Katello is not installed.

File: guides/doc-Provisioning_Guide/topics/Cloud-Amazon_EC2.adoc (hunk 1 above)
Comment (melcorr) [reply guides/doc-Provisioning_Guide/topics/Cloud-Amazon_EC2.adoc]: I will read yours and then I think we should add a standardized form of this sentence to the readme so we don't rewrite it the same way every time. 
If we write it identically, this should help with machine translation, or human translators can identify we have one sentence everywhere and do a sed on their translations.

File: guides/doc-Provisioning_Guide/topics/Cloud-Amazon_EC2.adoc (hunk 4 above)
Comment (melcorr) [reply guides/doc-Provisioning_Guide/topics/Cloud-Amazon_EC2.adoc]: Ha, this is why we are doing this. Thank you so much. I had no idea.

Comment (melcorr) [reply guides/doc-Provisioning_Guide/topics/Cloud-Amazon_EC2.adoc]: This is displayed as unchanged but if you look in the file I have made the changes.

**Comment by melcorr** [comment]
I think this is good to go, @lzap
Thanks for your review.

File: guides/doc-Provisioning_Guide/topics/Cloud-Amazon_EC2.adoc [hunk 5]
```diff
@@ -247,10 +271,20 @@ EOS
 . Click the *Templates* tab, and from the *Finish Template* list, select your finish template.
 . Navigate to *Hosts* > *Create Host* and enter the information about the host that you want to create.
 . Click the *Parameters* tab and navigate to *Host parameters*.
+ifeval::["{Build}" == "satellite"]
 . In *Host parameters*, click the *Add Parameter* button three times to add three new parameter fields. Add the following three parameters:
+endif::[]
+ifeval::["{Build}" == "foreman"]
+. If you have the Remote Execution plugin installed, in *Host parameters*, click the *Add Parameter* button two times to add two new parameter fields. If you use the Katello plugin, add a third parameter field. Add the following parameters:
+endif::[]
```
Comment (lzap) [inline guides/doc-Provisioning_Guide/topics/Cloud-Amazon_EC2.adoc]: I think this is sane way of doing things. Does not look so bad in the adoc source. Good.

**Pull request merged**